ts/sagas/wallet.ts

```diff
@@ -24,12 +77,161 @@ function* fetchTransactions(
 }
 
 function* fetchCreditCards(
-  loadCards: () => Promise<ReadonlyArray<CreditCard>>
+  loadCards: () => Promise<ReadonlyArray<Wallet>>
 ): Iterator<Effect> {
-  const cards: ReadonlyArray<CreditCard> = yield call(loadCards);
+  const cards: ReadonlyArray<Wallet> = yield call(loadCards);
   yield put(cardsFetched(cards));
 }
 
+const navigateTo = (routeName: string, params?: object) => {
+  return NavigationActions.navigate({ routeName, params });
+};
+
+function* paymentSaga(): Iterator<Effect> {
+  yield put(navigateTo(ROUTES.PAYMENT_SCAN_QR_CODE)); // start by showing qr code scanner
+
+  while (true) {
+    const action = yield take([
+      PAYMENT_INSERT_DATA_MANUALLY,
+      PAYMENT_SHOW_SUMMARY,
+      PAYMENT_CONFIRM_SUMMARY,
+      PAYMENT_PICK_PAYMENT_METHOD,
+      PAYMENT_CONFIRM_PAYMENT_METHOD,
+      PAYMENT_REQUEST_OTP,
+      PAYMENT_VERIFY_OTP,
+      COMPLETE_PAYMENT
+    ]);
+    if (action.type === COMPLETE_PAYMENT) {
+      break;
+    }
+    yield fork(handlePaymentActions, action);
+  }
+}
+function* handlePaymentActions(action: Action): Iterator<Effect> {
```

Comment:
instead of this huge function, I suggest to split the handlers for each action in their own generators - see for instance what @fpersico did here: https://github.com/teamdigitale/italia-app/blob/master/ts/sagas/messages.ts